ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/TransactionConsistencyValidatorSpec.scala

```diff
@@ -221,6 +230,28 @@ class ContractKeysValidatorSpec extends AnyWordSpec with Matchers {
         getTransactionRejectionReason(result).getInconsistent.getDetails
       rejectionReason should startWith("InconsistentKeys")
     }
+
+    "fail if a contract is not active anymore" in {
+      val globalCid = s"#$freshContractId"
+      val globalCreate = newCreateNodeWithFixedKey(globalCid)
+      val context = createCommitContext(
+        recordTime = None,
+        inputs = Map(
+          makeContractIdStateKey(globalCid) -> Some(
+            makeContractIdStateValue().toBuilder
+              .setContractState(
+                DamlContractState.newBuilder().setArchivedAt(Timestamp.getDefaultInstance)
+              )
+              .build()
+          )
+        ),
+      )
+      val builder = TransactionBuilder()
+      builder.add(archive(globalCreate, Set("Alice")))
+      val transaction = builder.buildSubmitted()
+      val result = validate(context, transaction)
+      result shouldBe a[StepStop]
```

Comment:
What is the expected rejection reason?

Reply:
Good catch, `Inconsistent`, adding this check.